### ISSUE TITLE
[Support] UpdateManager throwing 404

### ISSUE THREAD

**Issue by ChaseFlorell** [Squirrel/Squirrel.Windows#388]
I can't seem to figure this one out. I'm passing a valid URL to the UpdateManager, but I get a 404 when it's called.

![](https://i.imgur.com/oSKwwCX.png)

The problem is that I "think" I have everything correct.

Here's my update manager code.

        public static async Task Update()
        {
            // commenting out this using block makes the error go away... error comes from here.
            using (var mgr = new UpdateManager(ConfigurationManager.AppSettings["SquirrelUpdatePath"]))
            {
                await mgr.UpdateApp();
            }
        }

And the App.Config.

    <appSettings>
      
      <add key="SquirrelUpdatePath" value="http://mca-local.tcpl.ca/shellapplication/releases/RELEASES" />
    </appSettings>

And the IIS stuff is working

![](https://i.imgur.com/PlSeekZ.png)

The interesting thing is what shows up in the squirrel log.

> 2015-07-16 12:31:32> Program: Starting Squirrel Updater: --install .
> 2015-07-16 12:31:32> Program: Starting install, writing to C:\ProgramData\chase_florell\SquirrelTemp
> 2015-07-16 12:31:32> Program: About to install to: C:\ProgramData\chase_florell\MCAlocal
> 2015-07-16 12:31:32> CheckForUpdateImpl: Failed to load local releases, starting from scratch: System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\ProgramData\chase_florell\MCAlocal\packages\RELEASES'.
>    at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
>    at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, > FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
>    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
>    at Squirrel.Utility.LoadLocalReleases(String localReleaseFile)
>    at Squirrel.UpdateManager.CheckForUpdateImpl.<CheckForUpdate>d__28.MoveNext()
> 2015-07-16 12:31:32> CheckForUpdateImpl: Reading RELEASES file from C:\ProgramData\chase_florell\SquirrelTemp
> 2015-07-16 12:31:32> CheckForUpdateImpl: First run or local directory is corrupt, starting from scratch
> 2015-07-16 12:31:33> ApplyReleasesImpl: Writing files to app directory: C:\ProgramData\chase_florell\MCAlocal\app-0.1.1.1
> 2015-07-16 12:31:35> ApplyReleasesImpl: Squirrel Enabled Apps: []
> 2015-07-16 12:31:35> ApplyReleasesImpl: No apps are marked as Squirrel-aware! Going to run them all
> 2015-07-16 12:31:35> ApplyReleasesImpl: About to create shortcuts for MCA.exe, rootAppDir C:\ProgramData\chase_florell\MCAlocal
> 2015-07-16 12:31:35> ApplyReleasesImpl: Creating shortcut for MCA.exe => C:\Users\chase_florell\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\TransCanada\MCA.lnk
> 2015-07-16 12:31:35> ApplyReleasesImpl: About to save shortcut: C:\Users\chase_florell\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\TransCanada\MCA.lnk (target C:\ProgramData\chase_florell\MCAlocal\Update.exe, workingDir C:\ProgramData\chase_florell\MCAlocal\app-0.1.1.1, args --processStart MCA.exe)
> 2015-07-16 12:31:35> ApplyReleasesImpl: Creating shortcut for MCA.exe => C:\Users\chase_florell\Desktop\MCA.lnk
> 2015-07-16 12:31:35> ApplyReleasesImpl: About to save shortcut: C:\Users\chase_florell\Desktop\MCA.lnk (target C:\ProgramData\chase_florell\MCAlocal\Update.exe, workingDir C:\ProgramData\chase_florell\MCAlocal\app-0.1.1.1, args --processStart MCA.exe)
> 2015-07-16 12:31:35> ApplyReleasesImpl: Starting fixPinnedExecutables
> 2015-07-16 12:31:35> ApplyReleasesImpl: fixPinnedExecutables: oldAppDirectories is empty, this is pointless
> 2015-07-16 12:31:35> ApplyReleasesImpl: cleanDeadVersions: for version 0.1.1.1
> 2015-07-16 12:31:35> ApplyReleasesImpl: cleanDeadVersions: exclude folder app-0.1.1.1

but based on the error above, it says that it can't find the `packages\RELEASES` files... but it's there.

![](https://i.imgur.com/cvxeJXI.png)

What am I doing Wrong?

**Comment by ChaseFlorell** [comment]
I also ran a test, adding the Application Name to the UpdateManager, but it's having no affect.

            using (var mgr = new UpdateManager(ConfigurationManager.AppSettings["SquirrelUpdatePath"], "MCAlocal"))
            {
                await mgr.UpdateApp();
            }

**Comment by paulcbetts** [comment]
Your download key needs to point to the folder that RELEASES is in, not the file itself

**Comment by ChaseFlorell** [comment]
Yep, I just figured it out about 10 minutes ago. I did previously have it pointing to the folder, but I did __not__ have the Application Name in the UpdateManager. So the http path was a red herring.

My App name is `MCA` but I append the environment that it's being deployed to to the end of the app name so that I can run multiple versions of the app on my local workstation.

    MCALocal
    MCADev
    MCAQual
    MCA  // production

the problem is that even though I was pointing to the directory (before), I didn't have the `appName` parameter populated in the `UpdateManager` and therefore Squirrel was looking for an app called "MCA" during the update call, which didn't exist.

NOW, I've added a second app.config parameter called `AppName` that I update during deployment, and I use it during the update call. That way the update method knows the name of the nupkg file it's looking for.

This bit is not well documented.
